pester.go

```diff
@@ -135,8 +137,12 @@ type ContextLogHook func(ctx context.Context, e ErrEntry)
 // BackoffStrategy is used to determine how long a retry request should wait until attempted
 type BackoffStrategy func(retry int) time.Duration
 
+// RetryStrategy is used for determining when to retry request based on the response status code.
+// If this function returns true, attempt is considered successful and no more attempts are performed.
```

Comment:
Would it make sense to invert the logic here? i.e.
```suggestion
// If this function returns true, we consider that the request has failed and should be retried.
```
(also doing the appropriate code changes, of course)

Makes the code a bit more readable IMO. Currently you've got a `Retry` function which, when it returns `true`, we consider "no more retries" which is a bit counterintuitive. If you go ahead with this change I would also recommend renaming `Retry` member attribute to `ShouldRetry` to make it even more clearer.

Reply:
Retry function returning true when we consider "no more retries" seems counterintuitive to me too, I agree with Vedran.

Reply:
Yeah, counterintuitive, will switch it back.